instrumentation/http_instrumentation.go

```diff
@@ -62,15 +65,12 @@ type instrumentedHandlerFactory struct {
 }
 
 // NewHandler creates a new instrumented HTTP handler with the given extra labels and calling the "next" handlers.
+// If no extra labels are provided, we fetch them from request metadata.
 func (m instrumentedHandlerFactory) NewHandler(extraLabels prometheus.Labels, next http.Handler) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		// if no extra labels are provided check for them in the context
 		if len(extraLabels) == 0 {
-			if labels := r.Context().Value(ExtraLabelContextKey); labels != nil {
-				ctxLabels, ok := labels.(prometheus.Labels)
-				if ok {
-					extraLabels = ctxLabels
-				}
+			if labels := chiLabelParser(r); len(labels) != 0 {
```

Comment:
something to keep in mind is that moving this from being injectable to being hardcoded here makes the package unusable with any other routing framework